internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -305,8 +294,34 @@ func (c *HcpOpenShiftClusterResource) Normalize(out *api.HCPOpenShiftCluster) {
 	}
 }
 
-func (c *HcpOpenShiftClusterResource) ValidateStatic() error {
-	return nil
+func (c *HcpOpenShiftClusterResource) ValidateStatic(current api.VersionedHCPOpenShiftCluster, updating bool, method string) *arm.CloudError {
+	var normalized api.HCPOpenShiftCluster
+	var errorDetails []arm.CloudErrorBody
+
+	cloudError := arm.NewCloudError(
+		http.StatusBadRequest,
+		arm.CloudErrorCodeMultipleErrorsOccurred, "",
+		"Content validation failed on multiple fields")
+	cloudError.Details = make([]arm.CloudErrorBody, 0)
+
+	// FIXME Validate visibility tags by comparing the new cluster (c) to current.
+
+	c.Normalize(&normalized)
+
+	errorDetails = api.ValidateRequest(validate, method, &normalized)
+	if errorDetails != nil {
+		cloudError.Details = append(cloudError.Details, errorDetails...)
+	}
+
+	switch len(cloudError.Details) {
```

Comment:
this looks overly complicated, are we expecting eventually to have more than one error?

```
if len(cloudError.Details) == 0 {
  return nil
}
cloudError.CloudErrorBody = &cloudError.Details[0]
return cloudError
```

Reply:
Yes, the intent is to report _all_ validation errors in the response, not just the first one encountered.

The way this works is the logic anticipates multiple validation errors.  Multiple validation errors will look something like this:
```
{
  "Code": "MultipleErrorsOccurred",
  "Message": "Content validation failed on multiple fields",
  "Details": [
    {
      "Code": "InvalidRequestContent",
      "Message": "something about the value was bad",
      "Target": "properties.spec.the.bad.field"
    },
    ... additional errors for other fields ...
  ],
}
```

But if there's only a single validation error then the nesting isn't necessary, so it gets "promoted" into the parent section like this:
```
{
  "Code": "InvalidRequestContent",
  "Message": "something about the value was bad",
  "Target": "properties.spec.the.bad.field"
}
```

The complexity is just trying to improve readability for the end user.